File: languages/python/src/fatorial.py
# -*- coding:utf-8 -*-
# PYTHON 3 ONLY
#
# Algorítmo...
# Calculando o fatorial de um valor fornecido pelo usuário
# Será informado o valor fatorial
# Objetivo é calcular uma expressão muito usada no dia a dia de muitos profissionais, [fatorial], além, do [while] que é uma estrutura contendo laço de repetição

# Atribuindo o valor default
option = "s"

# Laço de repetição, enquanto o usuário escolher "SIM", ocorrerá a repetição
while (option == "s"):

    # Estrutura condicional, se o usuário escolher "SIM", vai ser dado o início do programa
    if (option == "s"):

        # Tipando as variáveis para o tipo número inteiro
        n = int(input("Digite um valor qualquer, \nInsira um número INTEIRO\n"))

         # Setando o valor default para o resultado e contador
        result = 1
        count = 1

        # Exibindo o resultado na tela
        print('O fatorial correspondente de {} é\n'.format(n))

        # Enquanto [count] for menor ou igual ao valor [n] inserido pelo usuário
        while (count <= n):

            # result = result * count
            result *= count

            # coutn = count + 1
            count += 1
        
            # Exibindo o resultado na tela
            print('{}\n'.format(result))

    # Fim da estrutura condicional, se o usuário escolher "NÃO", o programa será encerrado
    else:

        # Imprimindo na tela o resultado
        print('Você escolheu SAIR!\n')

    # Permite o usuário reiniciar automaticamente o LOOP, ou seja, o laço de repetição
    option = input("Deseja rodar o código novamente? \nPressione [s] para SIM \nOu \nPressione [n] para NÃO\n")

# Verifica se o usuário optou por sair, ao inserir um valor diferente de "s"
if (option != "s"):

    # Imprimindo na tela o resultado
    print('Você escolheu SAIR!\n')

    # Encerra o programa
    exit()

File: languages/dart/src/fatorial.dart
import 'dart:io';

void main() {
  String option = 's';

  while (option == 's') {
    if (option == 's') {
      print('Digite um valor inteiro positivo para calcular o fatorial:');
      int num = int.parse(stdin.readLineSync()!);

      if (num < 0) {
        print('Por favor, insira um número inteiro positivo.\n');
        continue;
      }

      // Setando o valor default para o resultado e contador
      int result = 1;
      int count = 1;

      // Exibindo o resultado na tela
      print('O fatorial correspondente de $num é\n');

      // Enquanto [count] for menor ou igual ao valor [n] inserido pelo usuário
      while (count <= num) {
        // result = result * count
        result *= count;
        print('Passo $count: $result');
        // coutn = count + 1
        count += 1;
      }

      // Exibindo o resultado na tela
      print('\nO fatorial de $num é $result\n');
    } else {
      print('Você escolheu SAIR!\n');
    }

    print(
      'Deseja rodar o código novamente?\n\nPressione [s] para SIM\nOu\nPressione [n] para NÃO\n',
    );
    option = stdin.readLineSync()!;
  }

  if (option != 's') {
    print('\nVocê escolheu SAIR!\n');
    exit(0);
  }
}
